src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java

```diff
@@ -133,6 +122,33 @@ public UserAuthenticationManager(final IUserDataManager database,
         this.setSecureCookies = isProduction;
     }
 
+    /**
+     * This method is used to extract user-identifying features of a request and return them in csv format.
+     * NOTE: The validity of the session token is not checked against the database.
+     *
+     * For interest the IP address format is described in the REMOTE_ADDR section of <a href="https://www.ietf.org/rfc/rfc3875">RFC 3875</>.
+     *
+     * @param request - http request from which to extract user identifying features.
+     * @return A string of comma-separated user identifying values from the request.
+     */
+    public String getUserIdentifierCsv(final HttpServletRequest request) {
+        String ipAddress = request.getRemoteAddr();
```

Comment:
On our servers (unlike local testing), this will always return the `10.x.x.x` IP address of the `isaac-router` proxy, since that is the other end of the connection. I recommend using `uk.ac.cam.cl.dtg.util.RequestIPExtractor::getClientIpAddr()`, which I built to do the right thing in all the messy cases we see!

Reply:
Nice! Hadn't seen that function, had wondered whether it would still know where it was coming from after the router/gateway.